### PR TITLE
Keycard authenticator can call ERT winout admin intervention

### LINKED ISSUE #4333: ERTs called via card-swiping do not actually call an ERT.
### Testmerges

#4246, #4224, #4063, #4314, #4332

### Description of the bug

Card-swiping to call an ERT only admin logs it and does nothing else.
Example: ADMIN LOG: x/(x) triggered and x/(x) confirmed event Emergency Response Team.

### What's the difference with what should have happened?

It should call an ERT.

### How do we reproduce this bug?

1. Just swipe your credit card in the CIC and watch as it takes your money without giving you the promised ERT!

### Issue Bingo

- [X] Issue could be reproduced at least once
- [X] Issue happened in a recent (less than 7 days ago) round
- [X] Couldn't find an existing issue about this (https://github.com/cmss13-devs/cmss13/issues)

### PR DESCRIPTION
**Pull request by Diegoflores31**
# About the pull request

Fixes #4333

the keycard authentication will now show an option to send a distress signal similar to the almayer control console that will allow to call the FIRST distress signal on the almayer winout the need for admin approval . ( only works on red alert)

# Explain why it's good for the game

Gives more options to lowpop command and provides a unique reward for using the authenticator in CIC instead of callling ERT remotely from lifeboats.

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
add: Adds a new way to send a Distress signal with the Keycard authenticator.
fix: Removes non functional "Call ERT" button in Keycard authenticator.
/:cl:

### CODE REVIEW

**Comment by Cthulhu80** [comment]
If that's the only time is_ert_blocked() is even used you might just want to shove that logic into call_ert() and simply return.

**Review by morrowwolf — CHANGES_REQUESTED** [review]
Auto ERTs isn't gonna fly.

It still needs admin approval.

**Comment by Diegoflores31** [comment]
> Auto ERTs isn't gonna fly.
> 
> It still needs admin approval.

but it only works for the first ERT . cannot be used more than once

**Review by morrowwolf — COMMENTED** [review]
If you keep it to post hijack and it works the same way for every way to call an ERT it can be a free one but it should give an opportunity for staff to cancel it and extra credit for a button somewhere to pre-toggle it off.

**Comment by Diegoflores31** [comment]
Ok

**Comment by Diegoflores31** [comment]
> If you keep it to post hijack and it works the same way for every way to call an ERT it can be a free one but it should give an opportunity for staff to cancel it and extra credit for a button somewhere to pre-toggle it off.

Changes applied:
-staff will have 15 seconds to cancel the ERT request . if it goes unanswered for more than that it will automatically approve
-If distress i called before hijack it will just call a standart ERT request,
-minor change applied ( you cant swipe for red alert if the ship is already on red alert)

File: code/datums/emergency_calls/emergency_call.dm
Comment (Zonespace27) [inline code/datums/emergency_calls/emergency_call.dm:24]: code doc

File: code/datums/emergency_calls/emergency_call.dm
Comment (Zonespace27) [inline code/datums/emergency_calls/emergency_call.dm:33]: code doc

**Review by Zonespace27 — CHANGES_REQUESTED** [review]
Follow cthulhu's reviews please

File: code/datums/emergency_calls/emergency_call.dm
Comment (Diegoflores31) [reply code/datums/emergency_calls/emergency_call.dm:24]: what code doc ?

Comment (realforest2001) [reply code/datums/emergency_calls/emergency_call.dm:24]: /// above proc instead of // next to it

File: code/modules/security_levels/keycard_authentication.dm
Comment (realforest2001) [inline code/modules/security_levels/keycard_authentication.dm:160]:
```suggestion

if(ert_dispatched)
		to_chat(usr, SPAN_WARNING("A distress beacon has already launched successfully!"))
		return
		
```

**Comment by Diegoflores31** [comment]
github web moment 💀

File: code/modules/security_levels/keycard_authentication.dm
Comment (fira) [inline code/modules/security_levels/keycard_authentication.dm:68]: remove the config flag in game_options.dm - it's now unused

**Review by fira — CHANGES_REQUESTED** [review]
looks good codewise

**Review by fira — APPROVED** [review]

**Review by harryob — APPROVED** [review]

**Comment by fira** [comment]
pls fix so we can merge it :(

**Comment by fira** [comment]
PR been on TM for a long time but players are complaining about ERT calls not going through, but i'm not sure if it's not just actually becuase most people don't know to look in the Keycard auth device or there is a real bug...

I'd like to request the PR to be double checked after all the new merges before it can be put on TM again and hopefully validated for good and merged

**Comment by fira** [comment]
Testing take 2 to confirm this effectively works as intended

**Comment by Diegoflores31** [comment]
There is currently a bug that some times your swipes wont register . I am almost 100% its because of table core . I will probably add some additional features today to make it more intuitive

**Review by Drulikar — CHANGES_REQUESTED** [review]
I think the issue with this is mostly just confusion/feedback to users.

Consider adding a sound/balloon_alert whenever the keycard_auth activates/times out (I believe that occurs here: https://github.com/cmss13-devs/cmss13/blob/master/code/modules/security_levels/keycard_authentication.dm#L116-L132

Also consider adding a message if a different user tries to authorize the same keycard_auth - or allow the same machine to both initiate and confirm a request if the triggerer/confirmer is a different user.

**Pull request closed without merging**
